src/ai.js

```diff
@@ -1,16 +1,651 @@
 const { Card } = require("./card");
 const { Game } = require("./game");
 const { Player } = require("./player");
-const { get } = require("./shared");
+const { get, set } = require("./shared");
+const lodash = require("lodash");
 
 /**
  * @type {Game}
  */
 let game = get();
 
+class SimulationAI {
+    /**
+     * @param {Player} plr 
+     */
+    constructor(plr) {
+        game = get();
+
+        /**
+         * @type {[[string, any]]}
+         */
+        this.history = [];
+
+        /**
+         * @type {boolean}
+         */
+        this.canAttack = true;
+
+        /**
+         * @type {Card | null}
+         */
+        this.on_card = null;
+
+        /**
+         * @type {SentimentAI}
+         */
+        this.backup_ai = new SentimentAI(plr);
+
+        /**
+         * @type {Player}
+         */
+        this.plr = plr;
+    }
+
+    /**
+     * Calculate the best move and return the result
+     * 
+     * @returns {Card | string} Result
+     */
+    chooseMove() {
+        // Makes a move in the simulation
+        // The ai is not even playing anything
+
+        /**
+         * @type {[Card | string, number]}
+         */
+        let best_move = [];
+
+        this.plr.hand.forEach(card => {
+            let simulation = this._createSimulation();
```

Comment:
^^^ This would be a massive performance gain, however might not work in some situations. If playing Card A prevents you from playing Card B, for example. I do think it is worth it.

Reply:
It doesn't quite work. Kept an implementation of it, commented out, in chooseMove